src/lib/utils/object-graph.ts

```diff
@@ -0,0 +1,172 @@
+import { v4 as uuidv4 } from 'uuid'
+
+/**
+ * Represents a serialized object structure where circular/repeated references
+ * are replaced with `{ __refId }` and the full objects are provided in the
+ * `include` map.
+ */
+export interface FlattenedGraph<T> {
+  data: T
+  include: Record<string, unknown>
+}
+
+/**
+ * A reference to an object stored in the `include` section of a
+ * `DereferencedData` structure.
+ */
+export interface Reference {
+  __refId: string
+}
+
+/**
+ * Serializes an object containing circular and/or repeated references into a
+ * JSON-safe format by extracting duplicates into an `include` map.
+ */
+export function flattenObjectGraph<T>(input: T): FlattenedGraph<T> {
```

Comment:
This runs at `O(n)` time and space complexity (where `n` is the number of nodes in the object graph)...but realistically, that's still `Θ(2n)` time and `Θ(3n)` space complexity (if I'm not mistaken...which I very well may be, given that we're storing by reference, not duplicate objects 🫠 ), which isn't ideal. I wasn't sure how to get that down.

Some of our object graphs are thousands of nodes (which is so silly), so if you can figure out how to optimize this, please let me know!

I don't _think_ this will cause meaningful performance degradation, but I also haven't benchmarked it. 🙈 I have an idea how I might do that, though... just gotta get [my entity-fetching CLI PR merged first](https://github.com/department-of-veterans-affairs/next-build/pull/1044). 😜